src/parser/ast_common.h

```diff
@@ -35,13 +35,20 @@ typedef struct {
 	char *property;
 } AST_Variable;
 
+typedef struct {
+	unsigned int minHops;
+	float maxHops;
```

Comment:
I'm guessing we're using a float for `maxHops` to provide simpler support for `*` (infinite) length paths? That makes sense, and is probably fine, but I wouldn't mind defining UINT_MAX as an infinity value either.

The smallest integer value that can't be represented as a float is 2^24 + 1, which is not a path length we reasonably need to be concerned about, so it's not a big deal - your call!